username_api.py

```diff
@@ -15,6 +15,11 @@
 
 cache = SimpleCache()
 
+session = r.Session()
+
+
+def get_with_session(link):
```

Comment:
no need to create separate function. we can directly call ``session.get`` wherever needed